### PR TITLE
Refactoring VALU/MFMA code into components

### PR DESCRIPTION
**Pull request by AlexBrownAMD**

### CODE REVIEW

File: Tensile/AsmUtils.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
+################################################################################
+# Copyright 2019-2020 Advanced Micro Devices, Inc. All rights reserved.
```
Comment (sdquiring) [inline Tensile/AsmUtils.py:2]: 2021?

File: Tensile/Components/ComputeStoreVgprs.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+################################################################################
+
+from ..Component import ComputeStoreVgprs
+from ..Common import globalParameters
+from ..AsmUtils import vectorStaticDivideAndRemainder, staticMultiply, vgpr, sgpr, inst
+
+class ComputeStoreVgprsVALU(ComputeStoreVgprs):
+    kernel = {"EnableMatrixInstruction": False}
+
+    ##############################################################################
```
Comment (sdquiring) [inline Tensile/Components/ComputeStoreVgprs.py:29]: Should be in a docstring at the class level.

File: Tensile/Components/ComputeStoreVgprs.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
+################################################################################
+# Copyright 2020 Advanced Micro Devices, Inc. All rights reserved.
```
Comment (sdquiring) [inline Tensile/Components/ComputeStoreVgprs.py:2]: 2021

File: Tensile/Components/LocalRead.py [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+                        # We only check the first one now
+                        # TODO: Handle vector, but need to take care the last one
+                        dbgVgprList = (dbgVgprList[1].split("]")[0]).split(':')
+                        dbgVgpr = "v[%s]"%dbgVgprList[0]
+
+                    localReadCode.addInst("s_waitcnt lgkmcnt(0)", "CheckValue1 wait for LDS read")
+                    if writer.archCaps["SeparateVscnt"]:
+                        localReadCode.addInst( "s_waitcnt_vscnt", "null", "0", "")
+
+                    if kernel["ProblemType"]["DataType"].isHalf():
```
Comment (sdquiring) [inline Tensile/Components/LocalRead.py:241]: Could be a candidate for another component, or maybe just adding literal value generation to the data type code?

File: Tensile/Components/ShiftVectorComponents.py [hunk 5]
```diff
@@ -0,0 +1,464 @@
...
+            So we do shift back by below alorithm.
+            1. check if M_size % vectorwidth != 0, return if == 0
+            2. decide which subgroup we need to shift, M_size(3) means 3/8 = group 0
+            3. decide which thread we need to shift, we have different groups of thread, (0-31) for first group, (32-63) for second group.
+            4. decide which shift block (subTile1) we want to shift. for ex [0-1], [1-2], we want to shift second subtile
+        """
+
+        kStr = ""
+
+        glvw                                             = tP["glvw"]
```
Comment (sdquiring) [inline Tensile/Components/ShiftVectorComponents.py:303]: Interesting formatting…

**Review by sdquiring — COMMENTED** [review]
Great progress!

**Review by sdquiring — APPROVED** [review]

**Comment by zaliu** [comment]
Your latest commit 93a47176 failed to build rocBLAS.

**Review by zaliu — APPROVED** [review]

**Comment by zaliu** [comment]
full rocblas-test passed on gfx908.

**Pull request merged**